README.md

```diff
@@ -20,212 +20,324 @@ To install shelf-lib-js for development, run the following commands:
     npm install
 
 
-Usage
------
+API
```

Comment:
Should "reference" and "artifact" be children of the "API" heading?

If so, you need to demote the other headings. Lots of minor changes.
If not (and I recommend this), perhaps "API" could be changed to "Library Interface" or some other wording in order to narrow the scope to just the thing that you get from `require()`?